sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanData.java

```diff
@@ -0,0 +1,386 @@
+   * @since 0.1.0
+   */
+  public abstract List<io.opentelemetry.trace.Link> getLinks();
+
+  /**
+   * Returns the {@code Status}.
+   *
+   * @return the {@code Status}.
+   * @since 0.1.0
+   */
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp}.
+   *
+   * @return the end {@code Timestamp}.
+   * @since 0.1.0
+   */
+  public abstract Timestamp getEndTimestamp();
+
+  /**
+   * A timed event representation.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvent {
+    /**
+     * Returns a new immutable {@code TimedEvent<T>}.
+     *
+     * @param timestamp the {@code Timestamp} of this event.
+     * @param event the event.
+     * @return a new immutable {@code TimedEvent<T>}
+     * @since 0.1.0
+     */
+    public static TimedEvent create(Timestamp timestamp, io.opentelemetry.trace.Event event) {
+      return new AutoValue_SpanData_TimedEvent(timestamp, event.getName(), event.getAttributes());
+    }
+
+    /**
+     * Returns the {@code Timestamp} of this event.
+     *
+     * @return the {@code Timestamp} of this event.
+     * @since 0.1.0
+     */
+    public abstract Timestamp getTimestamp();
+
+    /**
+     * Returns the name of this event.
+     *
+     * @return the name of this event.
+     */
+    public abstract String getName();
+
+    /**
+     * Gets the attributes for this event.
+     *
+     * @return the attributes for this event.
+     */
+    public abstract Map<String, AttributeValue> getAttributes();
+
+    TimedEvent() {}
+  }
+
+  /**
+   * Creates a new Builder for creating an SpanData instance.
+   *
+   * @return a new Builder.
+   * @since 0.1.0
+   */
+  public static Builder newBuilder() {
+    return new AutoValue_SpanData.Builder();
+  }
+
+  /**
+   * A {@code Builder} class for {@link SpanData}.
+   *
+   * @since 0.1.0
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    public abstract Builder traceId(TraceId traceId);
+
+    public abstract Builder spanId(SpanId spanId);
+
+    public abstract Builder traceFlags(TraceFlags traceFlags);
+
+    public abstract Builder tracestate(Tracestate tracestate);
+
+    /**
+     * The parent span id associated for this span, which may be null.
+     *
+     * @param parentSpanId the SpanId of the parent
+     * @return this
+     * @see SpanId
+     * @since 0.1.0
+     */
+    public abstract Builder parentSpanId(SpanId parentSpanId);
+
+    /**
+     * Set the resource associated with this span. Must not be null.
+     *
+     * @param resource the Resource that generated this span.
+     * @return this
+     * @see Resource
+     * @since 0.1.0
+     */
+    public abstract Builder resource(Resource resource);
+
+    /**
+     * Set the name of the span. Must not be null.
+     *
+     * @param name the name.
+     * @return this
+     * @since 0.1.0
+     */
+    public abstract Builder name(String name);
```

Comment:
for Builder methods we should use setName?

Reply:
That doesn't seem to be the current standard.  My preference would be to omit `set`.

Reply:
let's try to stick with set please. We use set consistently in this repo, if you want to make a change for that we should do it consistently on the entire repo

Reply:
👍